### DISCUSSION TITLE
Software usage

### DISCUSSION THREAD

**Discussion by rifi07** [OpenGammaProject/Open-Gamma-Detector#28] [Q&A] [answered]
Hi, I am using your project in order to make one of my own, but I am using an MKR ZERO board. And I would like to ask if it would be possible to know a bit how the program works, and what parts I would need just to do the detections as I would like to convert it into an MKR ZERO usable program. 

Thanks

**Comment by NuclearPhoenixx** [comment, marked as answer]
Hi,
Basically the most important function is the `eventInt` function that is tied to an interrupt. It triggers when the threshold is exceeded and enabled the ADC measurements with averaging and so on. You'd need to keep that, most of the other functions only deal with Pico related stuff or serial input/output.

The setup functions are basically self explanatory and the loop functions mostly only handle the serial output. I have no idea what the MKR ZERO can or cannot do, so you'll have to figure it out on your own. You could just try to compile it within the Arduino IDE and just remove bits that throw an error except for the `eventInt` function obviously.

Hope I could help at least a little bit.

**Reply by rifi07** [reply]
Ok, thanks

So just to know, what would I receive just by analog reading the AIN pin?

**Reply by NuclearPhoenixx** [reply]
You'll get whatever voltage the peak and hold circuit currently stores. This will be somewhere near <100 mV most of the time and sometimes larger when a peak has been generated by gamma-rays.

If you want to read it this way, you'll also have to manually reset it back to the original state everytime it registers a pulse. The (digital) reset pin has been defined in the sketch, I don't remember the exact pin number right now.
